.github/workflows/main.yml

```diff
@@ -67,12 +114,12 @@ jobs:
             path: /tmp/image
 
       - name: Import docker image
-        run: gunzip -c /tmp/image/research-template.tar.gz | docker load
+        run: docker load --input /tmp/image/research-template.tar.gz
 
       - name: Publish image
         run: |
-            echo ${{ secrets.GITHUB_TOKEN }} | docker login $REGISTRY -u ${{ github.actor }} --password-stdin
-            docker tag $IMAGE_NAME $PUBLIC_IMAGE_NAME:v0
-            docker tag $IMAGE_NAME $PUBLIC_IMAGE_NAME:latest
-            docker push $PUBLIC_IMAGE_NAME:v0
-            docker push $PUBLIC_IMAGE_NAME:latest
+            echo ${{ secrets.GITHUB_TOKEN }} | docker login "$REGISTRY" -u ${{ github.actor }} --password-stdin
+            docker tag "$IMAGE_NAME" "$PUBLIC_IMAGE_NAME:$IMAGE_VERSION"
+            docker tag "$IMAGE_NAME" "$PUBLIC_IMAGE_NAME:latest"
+            docker push "$PUBLIC_IMAGE_NAME:$IMAGE_VERSION"
+            docker push "$PUBLIC_IMAGE_NAME:latest"
```

Comment:
Are the quotes necessary? If so, then why?

Reply:
I typically use [`actionlint`](https://github.com/rhysd/actionlint) when working on GitHub Actions. `actionlint` in turn uses [`shellcheck`](https://github.com/koalaman/shellcheck) for linting the shell commands that are run in workflows.

The lack of double quotes for these variables results in [this warning](https://www.shellcheck.net/wiki/SC2086). Since I was using double quotes elsewhere, it seemed reasonable to add them throughout for consistency in the edited file.

Quotes are not _absolutely_ necessary given the specific values that are being used here; there are no spaces or filename expansions there. However, it's more difficult to reason that through for each variable, than simply making them safe by quoting.

We don't internally have any standards for shell scripts — maybe we should? In the absence of that, for example, [Google's style guide](https://google.github.io/styleguide/shellguide.html#quoting) agrees with `shellcheck`:

> Always quote strings containing variables, command substitutions, spaces or shell meta characters, unless careful unquoted expansion is required or it’s a shell-internal integer

Reply:
That's interesting. Thank you for replying! It would be worth mentioning this kind of thing in the commit message, as it makes the PR easier to review.

Reply:
Absolutely; I'll note the reasoning there.

Reply:
I reworded the commit message to add this.